Doc/library/stackless/tasklets.rst

```diff
@@ -511,3 +585,67 @@ state transitions these functions are roughly equivalent to the following
 
    def schedule_remove():
        stackless.current.next.switch()
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Tasklets and Context Variables
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. versionadded:: 3.7.6
+
+Version 3.7 of the |PPL| adds context variables, see module :mod:`contextvars`.
+Usually they are used in connection with
+:mod:`asyncio`, but they are a useful concept for |SLP| too.
+Using context variables and multiple tasklets together didn't work well in |SLP| versions 3.7.0 to
+3.7.5, because all tasklets of a given thread shared the same context.
+
+Starting with version 3.7.6 |SLP| adds explicit support for context variables.
+Design requirements were
+
+1. Be fully compatible with |CPY| and its design decisions.
+2. Be fully compatible with previous applications of |SLP|, which are unaware of context variables.
+3. Automatically share a context between related tasklets. This way a tasklet, that needs to set
+   a context variable, can delegate this duty to a sub-tasklet without the need to manage the
+   context of the sub-tasklet manually.
+4. Enable the integration of tasklet-based co-routines into the :mod:`asyncio` framework.
+   This is an obvious application which involves context variables and tasklets.
+
+Now each tasklet object has a private context attribute, which is either undefined (``NULL``) or a
+:class:`~contextvars.Context` object. The design goals have some consequences:
+
+* The active :class:`~contextvars.Context` object of a thread (as defined by the |PPL|)
+  is the context of the :attr:`~stackless.current` tasklet. This implies that a tasklet switch,
+  switches the active context of the thread.
+
+* In accordance with the design decisions made in :pep:`576` the context of a tasklet can't be
+  accessed directly, but you can use the method :meth:`tasklet.context_run` to run arbitrary code
+  in this context. For instance ``tasklet.context_run(contextvars.copy_context())`` returns a copy
+  of the context.
+  The attribute :attr:`tasklet.context_id` can be used to test, if two tasklets share the context.
+
+* A tasklet, whose context is undefined must behave identically to a tasklet, whose context is an
```

Comment:
I'm not sure why a tasklet may have an undefined (None) context.  Doesn't a thread always have a defined context?  Why doesn't the thread's main tasklet just have that context? How does a tasklet's state become undefined?

Reply:
Hi, my wording is not clear. In C a tasklet (and in C-Python a thread) has initially a NULL context. But this is invisible from Python, because Python transparently creates and sets an empty Context object, when you try to access `ts->context`, `tasklet->context` or if you set a ContextVar. I'll try to reword this section.

In an early version of my patch Python code could see a difference a NULL context and an empty context. That was a bad idea.